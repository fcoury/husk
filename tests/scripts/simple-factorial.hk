fn factorial(n: int) -> int {
    if n == 0 {
        1
    } else {
        n * factorial(n - 1)
    }
}

println!("Done defining factorial");
let result = factorial(3);
println!("factorial(3) = {}", result);