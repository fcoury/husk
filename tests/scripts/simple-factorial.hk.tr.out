Done defining factorial
factorial(3) =  6
